### PR TITLE
Update compatibility handling

### PR DESCRIPTION
**Pull request by mpalmi**
Updated compatibility handling and copyright bump. Tested on REL9_4_STABLE-master.

### CODE REVIEW

File: compatibility.h [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+#define _prev_hook \
+	prev_hook(pstmt, queryString, context, params, queryEnv, dest, completionTag)
+
+#define _standard_ProcessUtility \
+		standard_ProcessUtility(pstmt, queryString, context, params, queryEnv, dest, completionTag)
+
+# endif
+
+#include "utils/varlena.h"
+#define HAS_PSTMT
```
Comment (mpalmi) [inline compatibility.h]: this is cruft. it can be dropped.

Comment (mpalmi) [reply compatibility.h]: i'll hold off for further reviews and fix this in the respin with whatever else needs to be fixed.

File: compatibility.h [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+ *
+ * Introduces QueryCompletion struct
+ */
+#if PG_VERSION_NUM >= 130000
+static void PU_hook(PlannedStmt *pstmt, const char *queryString,
+					ProcessUtilityContext context, ParamListInfo params,
+					QueryEnvironment *queryEnv,
+					DestReceiver *dest, QueryCompletion *qc);
+#define _PU_HOOK \
+static void PU_hook(PlannedStmt *pstmt, const char *queryString, \
```
Comment (gitstashpop) [inline compatibility.h]: can you introduce a tab here to be consistent with the other macro definitions? Same applies to the rest of the macros below. It increases readability imo

File: compatibility.h [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+#endif /* 12+ */
+
+/*
+ * PostgreSQL version 10+
+ *
+ * - Introduces PlannedStmt struct
+ * - Introduces varlena.h
+ */
+#if PG_VERSION_NUM >= 100000
+# ifndef _PU_HOOK
```
Comment (gitstashpop) [inline compatibility.h:62]: is it possible for `_PU_HOOK` to ever be defined or is this leftover cruft from before?

File: set_user.c [hunk 4]
```diff
@@ -98,65 +43,28 @@
...
 static bool Block_AS = false;
-#endif
-
-#ifdef HAS_COPY_PROGRAM
-/* 9.3 & up */
-static bool Block_CP = false;
-#endif
-
 static bool Block_LS = false;
+static bool Block_CP = false;
```
Comment (gitstashpop) [inline set_user.c:62]: you can reduce your diff if you place Block_CP above Block_LS

**Review by gitstashpop — APPROVED** [review]
nits about formatting. LGTM assuming you tested on PG 9.6,10,11,12

File: set_user.c (hunk 4 above)
Comment (mpalmi) [reply set_user.c:62]: good eye

File: compatibility.h (hunk 3 above)
Comment (mpalmi) [reply compatibility.h:62]: yeah, the version numbers are all `if`, so this is like a fallthrough for a switch statement.

**Pull request merged**